shenanigan/metrics/core.py

```diff
@@ -0,0 +1,38 @@
+import numpy as np
+import tensorflow as tf
+from scipy.special import softmax
+from tensorflow.keras.applications.inception_v3 import preprocess_input
+from tensorflow.keras.models import load_model
+
+
+def inception_score(experiment_name, images, n_split=10, eps=1E-16):
+    """
+    adapted from: https://machinelearningmastery.com/how-to-implement-the-inception-score-from-scratch-for-evaluating-generated-images/
+    """
+    path = f"results/{experiment_name}/inception/model"
+    model = load_model(path)
+    processed = np.asarray((images + 1) * 255. / 2, np.uint8)
+    processed = preprocess_input(processed)
+    yhat = model.predict(processed, batch_size=1)
+    yhat = softmax(yhat, axis=1)
+    scores = list()
+    n_part = int(np.floor(images.shape[0] / n_split))
+    for i in range(n_split):
```

Comment:
I don't really understand why we have to split this up?

Reply:
original algorithm does, think it is to get a mean and std dev out of it. cross-validation-ish